py/warning.c

```diff
@@ -32,18 +32,23 @@
 
 #if MICROPY_WARNINGS
 
-void mp_warning(const char *msg, ...) {
+void mp_warning(const char *category, const char *msg, ...) {
```

Comment:
Types obviously won't work here as a first argument.  I guess this entire function would need to be configurable.